plugins/plugin-java/che-plugin-java-ext-lang-client/src/main/java/org/eclipse/che/ide/ext/java/client/navigation/service/JavaNavigationServiceImpl.java

```diff
@@ -237,23 +247,39 @@ public void getContent(
   }
 
   @Override
-  public Promise<ImplementationsDescriptorDTO> getImplementations(
-      Path project, String fqn, int offset) {
-    final String url =
-        appContext.getWsAgentServerApiEndpoint()
-            + "/java/navigation/implementations"
-            + "?projectpath="
-            + project.toString()
-            + "&fqn="
-            + fqn
-            + "&offset="
-            + offset;
-
-    return requestFactory
-        .createGetRequest(url)
-        .header(ACCEPT, APPLICATION_JSON)
-        .loader(loaderFactory.newLoader())
-        .send(unmarshallerFactory.newUnmarshaller(ImplementationsDescriptorDTO.class));
+  public Promise<ImplementationsDescriptor> findImplementations(
+      FindImplementationsCommandParameters params) {
+    return Promises.create(
+        (resolve, reject) -> {
+          requestTransmitter
+              .newRequest()
+              .endpointId(WS_AGENT_JSON_RPC_ENDPOINT_ID)
+              .methodName("java/navigation")
```

Comment:
"java/navigation" is not a good message name. "java/implementers" or the likes would be better.